app/main.py

```diff
@@ -1,8 +1,29 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name, age):
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+        self.name = name
+        self.age = age
+        self.spouse = None
+        Person.people[name] = self
+
+    def __repr__(self):
+
+        spouse_info = f", Spouse: {self.spouse.name}" if self.spouse else ""
+        return f"Person(Name: {self.name}, Age: {self.age}{spouse_info})"
+
+
+def create_person_list(people_data):
+    for person in people_data:
+        name = person["name"]
+        age = person["age"]
+        Person(name, age)
+
+    for person in people_data:
+        name = person["name"]
+        spouse_name = person.get("wife") or person.get("husband")
+        if spouse_name:
+            Person.people[name].spouse = Person.people.get(spouse_name)
```

Comment:
The code assumes that the spouse's name is unique and present in the `people_data`. If a spouse's name is not found in `Person.people`, this will result in a `None` value being assigned to `spouse`, which might not be the intended behavior. Consider adding error handling or validation to ensure that the spouse exists in the data.